src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+    register.add_run_env(PrintEnv)
+
+
+@impl
+def tox_add_core_config(core_conf, config):
+    if config.options.current_env:
+        config.options.default_runner = "current-env"
+        return
+
+    if config.options.print_deps_only:
+        warnings.warn(
+            "--print-deps-only is deprecated; use `--print-deps-to -`",
+            DeprecationWarning,
+        )
+        if not config.options.print_deps_to:
+            config.options.print_deps_to = sys.stdout
+        else:
+            raise RuntimeError(
+                "--print-deps-only cannot be used together " "with --print-deps-to"
+            )
+
+    if getattr(config.options.print_deps_to, "name", object()) == getattr(
+        config.options.print_extras_to, "name", object()
+    ):
+        raise RuntimeError(
+            "The paths given to --print-deps-to and --print-extras-to cannot be identical."
+        )
+
+    if config.options.print_deps_to or config.options.print_extras_to:
+        config.options.default_runner = "print-env"
+        return
+
+    # No options used - switch back to the standard runner
+    # Workaround for: https://github.com/tox-dev/tox/issues/2264
+    config.options.default_runner = "virtualenv"
+
+
+class CurrentEnv(PythonRun):
+    def __init__(self, create_args):
+        self._executor = None
+        self._installer = None
+        self._path = []
+        super().__init__(create_args)
+
+    @staticmethod
+    def id():
+        return "current-env"
+
+    @property
+    def _default_package_tox_env_type(self):
+        return None
+
+    @property
+    def _external_pkg_tox_env_type(self):
+        return None
+
+    @property
+    def _package_tox_env_type(self):
+        return None
+
+    @property
+    def executor(self):
+        if self._executor is None:
+            self._executor = CurrentEnvRunExecutor(self.options.is_colored)
+        return self._executor
+
+    def _get_python(self, base_python):
+        # TODO: Improve version check and error message
+        version_nodot = "".join(str(p) for p in sys.version_info[:2])
+        base_python = base_python[0]
+        if not base_python.endswith(version_nodot):
+            raise RuntimeError(
+                "Python version mismatch. "
+                f"Current version: {sys.version_info[:2]}, "
+                f"Requested environment: {base_python}"
+            )
+
+        return PythonInfo(
+            implementation=sys.implementation,
+            version_info=sys.version_info,
+            version=sys.version,
+            is_64=(platform.architecture()[0] == "64bit"),
+            platform=platform.platform(),
+            extra={"executable": Path(sys.executable)},
+        )
+
+    def create_python_env(self):
+        return None
+
+    def env_bin_dir(self):
+        return Path(sys.prefix) / "bin"
+
+    def env_python(self):
+        return sys.executable
+
+    def env_site_package_dir(self):
+        return Path(sys.prefix) / "lib"
```

Comment:
I think you're missing pyhtonx.y part. also This would not work on windows, instead you'd want to use sysconfig to find out things

Reply:
Have to check this. It worked well when tested manually.

Reply:
This might possibly end up read but unused?